modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -121,6 +121,47 @@ struct GInferBase {
     }
 };
 
+struct InOutInfo
+{
+    std::vector<std::string> in_names;
+    std::vector<std::string> out_names;
+};
+
+class GInferInputs
+{
+public:
+    cv::GMat& operator[](const std::string& name) { return in_blobs[name]; }
+    const std::unordered_map<std::string, cv::GMat>& getBlobs() const { return in_blobs; }
+
+private:
+    std::unordered_map<std::string, cv::GMat> in_blobs;
+};
+
+struct GInferOutputs
+{
+public:
+    GInferOutputs(std::shared_ptr<cv::GCall> call)
+        : m_call(std::move(call)), m_info(cv::util::any_cast<InOutInfo>(&m_call->params()))
+    {
+    };
+
+    cv::GMat at(const std::string& name)
+    {
+        auto it = out_blobs.find(name);
+        if (it == out_blobs.end()) {
+            m_call->kernel().outShapes.push_back(cv::GShape::GMAT);
+            int size = static_cast<int>(out_blobs.size());
```

Comment:
size is `size_t`

Reply:
Yes, but `GCall::yield(int)`

Reply:
The comment is about variable naming. Your `size` is not size, it is an index. `yield()` also gets index as a parameter.

Reply:
Got it, and naming already fixed